core/domain/suggestion_services.py

```diff
@@ -1826,3 +1828,805 @@ def update_question_suggestion(
+
+    Args:
+        suggestion: Suggestion. The suggestion domain object that is being
+            reviewed.
+        user_id: str. The ID of the corresponding submitter.
+        target_id: str. The ID of the target entity being suggested to.
+    """
+    topic_id = ''
+    exp_opportunity_dict = (
+        opportunity_services.get_exploration_opportunity_summaries_by_ids(
+            [target_id]))
+    exp_opportunity = exp_opportunity_dict[target_id]
+    # Ruling out the possibility of None for mypy type checking.
+    assert exp_opportunity is not None
+    topic_id = exp_opportunity.topic_id
+
+    content_plain_text = html_cleaner.strip_html_tags(
+        suggestion.change.translation_html)
+    content_word_count = len(content_plain_text.split())
+
+    translation_contribution_stat_model = (
+        suggestion_models.TranslationContributionStatsModel.get(
+            suggestion.change.language_code, user_id, topic_id
+        ))
+
+    if translation_contribution_stat_model is None:
+        suggestion_models.TranslationContributionStatsModel.create(
+            language_code=suggestion.change.language_code,
+            contributor_user_id=user_id,
+            topic_id=topic_id,
+            submitted_translations_count=1,
+            submitted_translation_word_count=content_word_count,
+            accepted_translations_count=0,
+            accepted_translations_without_reviewer_edits_count=0,
+            accepted_translation_word_count=0,
+            rejected_translations_count=0,
+            rejected_translation_word_count=0,
+            contribution_dates=[
+                datetime.datetime.strptime(
+                    str(
+                        suggestion.last_updated.utcnow().date().isoformat()),
+                        '%Y-%m-%d'
+                    ).date()
+            ]
+        )
+    else:
+        translation_contribution_stat = (
+            create_translation_contribution_stats_from_model(
+                translation_contribution_stat_model))
+
+        suggestion_is_accepted = (
+        suggestion.status == suggestion_models.STATUS_ACCEPTED)
+
+        suggestion_is_rejected = (
+            suggestion.status == suggestion_models.STATUS_REJECTED)
+        (
+            translation_contribution_stat
+            .accepted_translations_count
+        ) += 1 * int(suggestion_is_accepted)
+        (
+            translation_contribution_stat
+            .accepted_translations_without_reviewer_edits_count
+        ) += 1 + int(suggestion.edited_by_reviewer)
+        translation_contribution_stat.accepted_translation_word_count += (
+            content_word_count) * int(suggestion_is_accepted)
+        translation_contribution_stat.rejected_translations_count += 1 * int(
+            suggestion_is_rejected)
+        (
+            translation_contribution_stat.rejected_translation_word_count
+        ) += content_word_count * int(
+            suggestion_is_rejected)
```

Comment:
I moved this into a separate function `get_incremental_translation_contribution_stats_at_review` and I don't think we need condition check in that function because it is a very simple function.